### ISSUE TITLE
Validate async fails silently

### ISSUE THREAD

**Issue by simon-p-r** [zaggino/z-schema#103]
Hi

Just found an issue where z-schema fails silently when trying to validate an invalid schema object in async mode.  Schema is here and failed due to empty enum array;

````JS
{ required: [ 'recType', 'luValue' ],
  properties: 
   { luValue: { type: 'string' },
     useAsIs: { type: 'integer' },
     valueText: { type: 'string' },
     sequence: { type: 'integer', minimum: 0 },
     description: { type: 'string' },
     valueInt: { type: 'integer' },
     recType: { type: 'string', enum: [] },
     recProps: { type: 'object', properties: {}, additionalProperties: false },
     control: { '$ref': 'control' } },
  additionalProperties: false,
  type: 'object',
  title: 'lookup schema for bankCodeType' }
````
The schema was failing on line 131 - 133 as being returned as sync function before detected if function was called async (line 138) which meant process hanged due to callback not being triggered due to no return value.  I can try to do a PR and add test if necessary. 

Thanks
Simon

**Comment by zaggino** [comment]
Fixed and published as `z-schema@3.9.4`

**Comment by simon-p-r** [comment]
Thanks for quick response
